ParabolInc/parabol
# feat: push meeting templates to CDN

# Description

Fixes #8070

- Template illustrations get bundled with the server assets
- a script gets called predeploy which looks for those assets & pushes them to a CDN if they don't exist
- it works for both S3 and local image hosting
- i had to bump the aws-sdk. It was reeeeeeeally old & calling `headObject` was sometimes failing
- After this is done we can start adding these URLs to rows in the DB

## Demo

Nothing to see :-(

## Testing scenarios

- run `yarn build` and look at /dist/templates
- run `node dist/migrateImages` and see the files have been pushed to S3 (e.g. https://action-files.parabol.co/development/store/Organization/aGhostOrg/template/gladSadMadTemplate.png)
- change your .env `FILE_STORE_PROVIDER='local'` and re-run the above 2 commands. see the files live in /self-hosted
- change your user avatar, just to make sure it works
